frontend/scripts/react-components/dashboard-element/dashboard-widget/table-modal/table-modal.selectors.js

```diff
@@ -0,0 +1,149 @@
+import { createSelector } from 'reselect';
+import isEmpty from 'lodash/isEmpty';
+import { format } from 'd3-format';
+import { CHART_TYPES } from 'constants';
+
+const getMeta = (state, { meta }) => meta || null;
+const getData = (state, { data }) => data || null;
+const getChartType = (state, { chartType }) => chartType || null;
+const returnNameArrayIfNotEmpty = value =>
+  isEmpty(value) ? null : Object.values(value).map(v => v.name);
+const getCountriesPanel = state =>
+  returnNameArrayIfNotEmpty(
+    state.dashboardElement && state.dashboardElement.countriesPanel.activeItems
+  );
+const getCommoditiesPanel = state =>
+  returnNameArrayIfNotEmpty(
+    state.dashboardElement && state.dashboardElement.commoditiesPanel.activeItems
+  );
+const getSourcesPanel = state =>
+  returnNameArrayIfNotEmpty(
+    state.dashboardElement && state.dashboardElement.commoditiesPanel.activeItems
+  );
+
+const getVariableColumnName = createSelector(
+  [getMeta],
+  meta => (meta.info.node_type === 'COUNTRY' ? 'DESTINATION' : meta.info.node_type)
+);
+
+const hasNContIndicator = createSelector(
+  [getMeta],
+  meta => meta.info.filter.ncont_attribute
+);
+const hasMultipleYears = createSelector(
+  [getMeta],
+  meta => meta.info.years.end_year
+);
+const hasVariableColumn = createSelector(
+  [getChartType, getVariableColumnName, hasMultipleYears, hasNContIndicator, getSourcesPanel],
+  (chartType, variableColumnName, _hasMultipleYears, _hasNContIndicator, sourcesPanel) =>
+    !(!sourcesPanel && _hasNContIndicator && _hasMultipleYears) &&
+    !(!sourcesPanel && chartType === 'bar' && _hasMultipleYears) &&
+    !(chartType === 'pie') &&
+    variableColumnName
+);
+
+const getIndicatorColumnName = (meta, type) => {
+  const unit = type === 'cont' && (meta.xAxis.suffix || meta.yAxis.suffix);
+  return `${meta.info.filter[`${type}_attribute`].toUpperCase()}${unit ? ` (${unit})` : ''}`;
```

Comment:
I generally don't like using uppercase strings, prefer having the CSS style it. Mainly because if you need to update to capitalised you have to track this transform and change it back.

Reply:
I think is even more obscured in the CSS 😟

Reply:
IMO CSS should take care of presentation, values shouldn't come in a presentational way.

Reply:
In this specific case is quite difficult as we will have to add a span and that will be ugly as this is a selector. Not all columns will have unit and that will add much more code to the component. Do you mind if we keep it like this or you have some alternative?

Reply:
You could return an object instead of a string. I do think that this is important.

`{ name: ‘trade volume’, unit: ‘t’ }`